akd/src/utils.rs

```diff
@@ -9,36 +9,42 @@
 // 2. For each node in current_nodes set, check if each child is in prefix hashmap
 // 3. If so, add child label to batch set
 
-use crate::{NodeLabel, EMPTY_LABEL, EMPTY_VALUE};
-use std::collections::HashSet;
+use crate::{Node, EMPTY_LABEL, EMPTY_VALUE};
 
-// Builds a set of all prefixes of the input labels
-pub(crate) fn build_prefixes_set(labels: &[NodeLabel]) -> HashSet<NodeLabel> {
-    let mut prefixes_set = HashSet::new();
-    for label in labels {
-        for len in 0..(label.get_len() + 1) {
-            prefixes_set.insert(label.get_prefix(len));
-        }
+/// A node with node.label.label_len = 256. Used a wrapper around a [Node] in order to
+/// help with preloading nodes during directory publish.
+#[derive(Eq)]
+pub(crate) struct MaxLengthNode(Node);
+
+impl PartialEq for MaxLengthNode {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.label.label_val == other.0.label.label_val
```

Comment:
nit: do we not support eq on a `NodeLabel` directly? Do we need to compare the inner values rather than just 
```rust
self.0 == other.0
```?

Reply:
Good point! Actually not relevant anymore since I decided to get rid of `MaxLengthNode` altogether.